### PR TITLE
Refactor code, add tests

### LINKED ISSUE #12: Schedule display
Heads up on Wednesday
Week number 4, two pairs should be displayed.
image
In the meantime:
image
What is most confusing to me is that the error occurred only with the reply button when I manually select the week number "Wednesday".
With the command `/week_schedule` there is no error, it shows two pairs:
image
After outputting the full schedule `/week_schedule` this error is no longer present, i.e. when manually selecting a day everything is displayed correctly.
image
image

### PR DESCRIPTION
**Pull request by Ilanaya**
Resolves #12 
Sorry, I didn't have time to check the issue. I couldn't reproduce what you described but I added the test to ensure it works as expected.

I also made some changes to the code as navigating and understanding it was quite hard for me

### CODE REVIEW

File: src/components/handlers/scheduleHandler.js [hunk 1]
```diff
@@ -56,31 +49,25 @@ export const getLessonType = (event, currentWeek) => {
...
-	// Find information about the lessons for the specified day
-	const lessonToday = dataBase.find(({ day, events }) => day === targetDay && events)
-	// If no lesson information is found for the specified day, return an error message
-	if (!lessonToday) return MESSAGE
-	// Get the current day of the week
-	const getNumberWeek = getWeek(new Date())
-	const currentWeek = shouldShiftWeek ? getNumberWeek + 1 : getNumberWeek
-	// Format lesson information for the specified day
+// eslint-disable-next-line max-len
+export const getLessonsInfo = (targetDay, shouldShiftWeek = false, scheduleData = dataBase, weekNumber = getWeekNumber(new Date())) => {
```
Comment (Ilanaya) [inline src/components/handlers/scheduleHandler.js:55]: I made this change at least to be able to test this function. In your implementation, you had a lot of data that came to function from the outer scopes (`dataBase` access) or was defined by the environment ( `new Date()` call), which made it impure and hard to test.

File: tests/scheduleHandler/getLessonsInfo.spec.js [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+					lection: [],
+					practical: [3, 5, 7],
+					description: 'Дегтярёва В.В.',
+					audience: 'ЛК - 402',
+				},
+			],
+		},
+	]
+	it('Works for weekday', () => {
+		expect(getLessonsInfo(WEEK_DAY, false, MOCK_SCHEDULE_DATA, 4)).toMatchSnapshot()
```
Comment (Ilanaya) [inline tests/scheduleHandler/getLessonsInfo.spec.js:48]: Snapshot testing is not the best practice but for the sake of simplicity I made it there.

**Review by Ilanaya — COMMENTED** [review]
As you can see, I deleted some of the comments. Most of them are just distracting and do not provide additional value.

**Comment by Ilanaya** [comment]
Please recheck everything is working before merging. I checked myself and ran tests but I can't be sure I didn't break something

**Comment by mkangoo** [comment]
Hello, thanks for improving the code. I tested the bot locally, this merge is safe. However, I must report that I noticed the following: after restarting the project on the server, the errors disappear, I also could not reproduce this error locally. I think I should look answer in [`amvera`](https://id.amvera.ru/auth/realms/amvera/protocol/openid-connect/auth?client_id=amvera-web&redirect_uri=https%3A%2F%2Fcloud.amvera.ru%2Fauth%2Fcallback&response_type=code&scope=openid&state=be501a21705f4f3ebb7f4028ae97b5a6&code_challenge=hdvMz_3N0E0uO6VP1lRFuW1pCORMty-ihtBdl13hKN0&code_challenge_method=S256&response_mode=query) support.

**Pull request merged**